clients/drcachesim/drpt2trace/drpt2trace.cpp

```diff
@@ -213,25 +234,13 @@ static droption_t<unsigned long long> op_sb_kernel_start(
  */
 
 static void
-print_results(IN instrlist_autoclean_t &ilist)
+print_results(IN instrlist_autoclean_t &drir, IN std::vector<trace_entry_t> &entries)
 {
-    if (ilist.data == nullptr) {
-        std::cerr << "The list to store decoded instructions is not initialized."
-                  << std::endl;
-        return;
-    }
-    instr_t *instr = instrlist_first(ilist.data);
-    uint64_t count = 0;
-    while (instr != NULL) {
-        count++;
-        instr = instr_get_next(instr);
-    }
-
     if (op_print_trace.specified()) {
         /* Print the disassemble code of the trace. */
-        instrlist_disassemble(GLOBAL_DCONTEXT, 0, ilist.data, STDOUT);
+        instrlist_disassemble(GLOBAL_DCONTEXT, 0, drir.data, STDOUT);
     }
-    std::cout << "Number of Instructions: " << count << std::endl;
+    std::cout << "Number of Instructions: " << entries.size() << std::endl;
```

Comment:
Once we add encoding entries this will not be the instruction count.  Maybe better to keep the count loop or change this to say "entries" instead of instructions or something.